pkg/kv/dist_sender.go

```diff
@@ -1230,26 +1229,19 @@ func (ds *DistSender) sendToReplicas(
 					// was already received, we must return an ambiguous commit
 					// error instead of returned error.
 					log.ErrEventf(ctx, "application error: %s", call.Reply.Error)
-					timer := time.NewTimer(defaultPendingRPCTimeout)
-					defer timer.Stop()
 					// If there are still pending RPC(s), try to wait them out.
-					for timedOut := false; pending > 0 && !timedOut; {
-						select {
-						case pendingCall := <-done:
-							pending--
-							if err := pendingCall.Err; err != nil {
-								if grpc.Code(err) != codes.Unavailable {
-									ambiguousResult = true
-								}
-							} else if pendingCall.Reply.Error == nil {
-								return pendingCall.Reply, nil
+					for ; pending > 0; pending-- {
```

Comment:
There's actually no point in waiting for these pending RPCs unless `haveCommit` is true. Let's change this line to:

`for ; haveCommit && pending > 0; pending-- {`